Feature/edit section

TkalyaRM opened the pull request:
[Создание центральная секции (логотип + проект) (10 hours)](https://tree.taiga.io/project/titkov2002-websequencer/task/31)

muslldev committed:
fix: remove the possibility of substituting garbage in key "accessToken"

TkalyaRM committed:
added logo and project name

TkalyaRM committed:
added logo and project name

TkalyaRM committed:
added editable project name

TkalyaRM committed:
fix: project name input keyboard

muslldev committed:
feat: add API mock tests

- Mock username
- Mock token

TkalyaRM committed:
added toast and zod

TkalyaRM committed:
added toast and zod

muslldev committed:
fix(toast): remove Toaster's default options and change toast's position and styling

muslldev committed:
fix(toast): remove Toaster's default options and change toast's position and styling

muslldev committed:
fix: decomposite work with project's name

muslldev committed:
fix: decomposite work with project's name

muslldev commented on src/app/App.tsx:
```diff
@@ -1,14 +1,16 @@
       <Router />
       <SoundManager />
       <LiveMIDIInput />
+      <Toaster />
```
Были дефолтные настройки для тостов,но теперь они в кастомном хуке

muslldev commented on src/layouts/InstrumentLayout/InstrumentLayout.scss:
```diff
@@ -1,15 +1,18 @@
-    height: 92%;
+    height: 80%;
   }
 
```
Решили проблему с высотами хэддера и сетки для музыки. Теперь хэддер не улетает вверх

vIad17 replied:
сейчас если пользователь незалогинен, то появляется слишком много пустого места.. Так что нужно отойти от фиксированной высоты и сделать её флексами. Не разберётесь с вёрсткой - обращайтесь

<img width="1919" height="255" alt="image" src="https://github.com/user-attachments/assets/b0cc7654-f6c4-4849-b79b-f945ac82de17" />

muslldev commented on src/layouts/InstrumentLayout/components/Header/Header.scss:
```diff
@@ -1,11 +1,12 @@
   color: #fff;
   display: flex;
-  align-items: center;
+  align-items: start;
```
Лого с названием должно было так позиционироваться. Чтобы элементы прилипли к верхней границе родителя

muslldev commented on src/layouts/InstrumentLayout/components/Header/Header.tsx:
```diff
@@ -185,86 +204,116 @@ const Header = ({ className = '' }: HeaderProps) => {
-                  Tone.Transport.start();
-                  dispatch(setIsPlaying(true));
-                }}
+          {localStorage.getItem('accessToken') && (
```
Тут как по ТЗ. Если токен будет,то авторизован и мы показываем лого и название проекта

vIad17 replied:
в будущем лого и название проекта нужно будет подвязать на id проекта. Мол если id есть, показываем. Если нет, скрываем. На проект может зайти и не залогиненный пользователь

muslldev reviewed (commented):
В принципе ок. Начало таски положено. Дальше работаем :)

vIad17 commented on src/layouts/InstrumentLayout/InstrumentLayout.scss:
```diff
@@ -1,15 +1,18 @@
 .instrument-layout {
   width: 100vw;
   height: 100dvh;
+  display: flex;
+  flex-direction: column;
```
из-за этих двух строк слетели z-index-ы, из-за чего модалка скрывается за самим полем секвенсора

<img width="592" height="229" alt="image" src="https://github.com/user-attachments/assets/6ba9cfeb-813c-4171-ac30-2be6489bd53e" />

vIad17 commented on src/layouts/InstrumentLayout/components/Header/Header.tsx:
```diff
@@ -185,86 +204,116 @@ const Header = ({ className = '' }: HeaderProps) => {
-                  className="header__icon header__icon_stop"
-                />
-              </button>
+                {isEditing ? (
```
хотел бы похвалить инпут. Очень круто сделан, очень удобный. Лайк!

<img width="323" height="64" alt="image" src="https://github.com/user-attachments/assets/9519933a-78cd-4663-95ed-94e1521fced6" />

vIad17 commented on src/shared/icons/svg/logo.svg:
это уже png либо jpeg. Svg, как я вижу рисует картинку чисто по пикселям, так что в векторном формате уже нет смысла

vIad17 reviewed (commented):
в целом гут, но есть замечания

muslldev committed:
fix: move getUserInfo in separate useEffect

muslldev committed:
fix: add PO's wishes 🔧

- Add yarn.lock in .gitignore
- Add Mock Tests instruction in README.md
- Add opportunity of compilation TS worker to JS worker

muslldev committed:
fix: add .env with dev settings and remove comments in TS MSW config

TkalyaRM committed:
fix: modal z-index stacking issue

TkalyaRM committed:
fix: replace SVG with PNG

TkalyaRM committed:
fix: sequencer fills available space

muslldev committed:
fix: remove repeated logo's import

TkalyaRM committed:
feat: add project mocks

muslldev committed:
feat: add mock requests with decomposition

muslldev committed:
fix(logo): return main logo in tab

vIad17 commented on src/layouts/InstrumentLayout/components/Header/Header.scss:
```diff
@@ -1,27 +1,86 @@
+  }
+
+  &__project-name {
+    font-family: var(--lato);
```
??

vIad17 commented on src/layouts/InstrumentLayout/components/Header/Header.tsx:
```diff
@@ -196,86 +220,118 @@ const Header = ({ className = '' }: HeaderProps) => {
+                  <Icon
+                    icon={IconType.Play}
+                    interactable
+                    className="header__icon header__icon-start"
+                  />
+                </button>
+                <button className="header__button" onClick={pauseMusic}>
+                  <Icon
+                    icon={IconType.Pause}
+                    interactable
+                    className="header__icon header__icon-pause"
+                  />
+                </button>
+                <button className="header__button" onClick={stopMusic}>
+                  <Icon
+                    icon={IconType.Repeat}
+                    interactable
+                    className="header__icon header__icon_stop"
```
микроопечатка
header__icon-stop

TkalyaRM committed:
Merge branch 'main' into feature/edit-section

vIad17 commented:
баг:
если нажать на инпут с именем проекта и сразу же нажать вне этого инпута (ничего не вводя), то стирается имя проекта. Ожидается поведение, что имя проекта останется неизменным
